File: artifacts/mimalloc/decompose/src/mi_thread_data_t.c
typedef struct mi_thread_data_s
{
  mi_heap_t heap;
  mi_tld_t tld;
  mi_memid_t memid;
} mi_thread_data_t
/*** DEPENDENCIES:
typedef struct mi_memid_s
{
  union 
  {
    mi_memid_os_info_t os;
    mi_memid_arena_info_t arena;
  } mem;
  bool is_pinned;
  bool initially_committed;
  bool initially_zero;
  mi_memkind_t memkind;
} mi_memid_t
----------------------------
None
----------------------------
***/

File: artifacts/mimalloc/manual_debug/src/mi_thread_data_t.rs
use crate::*;
use std::sync::atomic::AtomicUsize;



pub struct MiThreadDataS {
    pub heap: MiHeapT,
    pub tld: MiTldT,
    pub memid: MiMemidS,
}

pub type MiThreadDataT = Option<Box<MiThreadDataS>>;

